### PR TITLE
mavlink: STATUSTEXT directly use mavlink_log subscription

### PR DESCRIPTION
**Pull request by dagar**
- mavlink STATUSTEXT stream directly subscribe to ORB_ID(mavlink_log) with increased queue depth
 - move STATUSTEXT stream to separate header

It's cheaper to have a single mavlink_log buffer at the uORB level than to have an additional ringbuffer in every mavlink instance.

### CODE REVIEW

**Comment by dagar** [comment]
Saves somewhere around 1.5-2 kB of memory on default fmu-v4 (3 mavlink instances).

File: src/drivers/camera_capture/camera_capture.cpp [hunk 1]
```diff
@@ -295,13 +292,6 @@ CameraCapture::reset_statistics(bool reset_seq)
 int
 CameraCapture::start()
 {
-	/* allocate basic report buffers */
-	_trig_buffer = new ringbuffer::RingBuffer(2, sizeof(_trig_s));
```
Comment (MaEtUgR) [inline src/drivers/camera_capture/camera_capture.cpp:299]: https://github.com/PX4/PX4-Autopilot/commit/faf535b040429bd566e3fddbbda6c8a5d09776c9#diff-21b8564105b699e09127da22888e7b35b845e53be24c30a6f921ea1f9e82f821L122
😅 👍

**Review by MaEtUgR — APPROVED** [review]
Looks like a very nice cleanup! Probably originally the buffer is there because there was no queue, right?
Just for easier reviewability, I'd suggest moving to `streams/` in a separate commit from the functional changes next time.
I don't see any catches.

**Comment by dagar** [comment]
> Probably originally the buffer is there because there was no queue, right?

Originally this RingBuffer was used all over the place in drivers (eg data from /dev/accelX) and we didn't have queuing at the uORB level.

**Pull request merged**